src/Microsoft.Diagnostics.Monitoring/Contracts/Metric.cs

```diff
@@ -19,7 +19,16 @@ public enum MetricType
 
     public class Metric
     {
-        public Metric(DateTime timestamp, string metricNamespace, string name, string displayName, string unit, double value, IReadOnlyList<string> dimNames, IReadOnlyList<string> dimValues, MetricType metricType = MetricType.Avg)
+        public Metric(DateTime timestamp,
+            string metricNamespace,
+            string name,
+            string displayName,
+            string unit,
+            double value,
+            MetricType metricType,
+            float interval,
```

Comment:
Can we use use `int` for the type of `interval`? interval is represented as float internally in CLR but it's only specifiable in terms of integer using Microsoft.Diagnostics.NETCore.Client library and other user-facing APIs that let you turn on counters.

Reply:
Note I'm extracting this value from IntervalSec from the counter payload. This is the real time difference rather than the declared integer value. I can extract the real integer interval as well, but the intent was to use this value to do the rate calculation.